src/com/google/javascript/jscomp/SourceFile.java

```diff
@@ -359,13 +359,13 @@ public String toString() {
   private static final String JAR_URL_PREFIX = "jar:file:";
 
   private static boolean isZipEntry(String path) {
-    return path.contains(".zip!" + File.separatorChar) && (path.endsWith(".js") || path.endsWith(".js.map"));
+    return path.contains(".zip!" + File.separator) && (path.endsWith(".js") || path.endsWith(".js.map"));
   }
 
   @GwtIncompatible("java.io.File")
   private static SourceFile fromZipEntry(String zipURL, Charset inputCharset) {
     checkArgument(isZipEntry(zipURL));
-    String[] components = zipURL.split(Pattern.quote(BANG_SLASH.replace('/',  File.separatorChar)));
+    String[] components = zipURL.split(Pattern.quote(BANG_SLASH.replace("/",  File.separator)));
```

Comment:
Instead of doing a replace, can you fix the `BANG_SLASH` definition to

`private static final String BANG_SLASH = "!" + File.separatorChar;`?

Reply:
This was also my first idea, but BANG_SLASH is used 3 times, but only one is windows (path) dependend. Only using another constant (like BANG_SLASH_FS_SPECIAL) would be a way to go, but I think this more confusing.